src/compiler/checker.ts

```diff
@@ -12643,6 +12643,22 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             });
             return concatenate(sig.parameters.slice(0, restIndex), restParams);
         }
+
+        function getUniqAssociatedNamesFromTupleType(type: TupleTypeReference) {
+            const associatedNamesMap = new Map<__String, number>();
+            return map(type.target.labeledElementDeclarations, labeledElement => {
+                const name = getTupleElementLabel(labeledElement);
+                const prevCounter = associatedNamesMap.get(name);
+                if (prevCounter === undefined) {
+                    associatedNamesMap.set(name, 1);
+                    return name;
+                }
+                else {
+                    associatedNamesMap.set(name, prevCounter + 1);
+                    return `${name}_${prevCounter}` as __String;

```

Comment:
Probably not for this PR, but I feel like I've seen this sort of thing a few times before, e.g. for renaming type variables in emit, other parameter name generation. I wonder if we can deduplicate this.

Reply:
Hm, also, we do have existing code for this sort of thing, except it starts at 0.

But, type variable renamings start at `1`. Oops. (I feel like starting at 1 is correct, but maybe that's because I was more recently working on the type variable code.)